recipe/meta.yaml

```diff
@@ -27,13 +27,13 @@ requirements:
     - pip
   run:
     - python >=3.8
-    - dask-core ==2023.1.1
-    - distributed ==2023.1.1
+    - dask-core ==2023.3.2
+    - distributed ==2023.3.2.1
     - dask ==2023.1.1
     - pynvml >=11.0.0,<11.5
-    - numpy >=1.18.0
+    - numpy >=1.21
     - numba >=0.54
-    - pandas >=1.0
+    - pandas >=1.3,<1.6.0dev0
```

Comment:
We may need a repodata hot-fix to add this to old versions

Reply:
Is there a relevant issue to link to on why we're doing this? I see https://github.com/rapidsai/dask-cuda/pull/1139 is where the max version constraint got introduced, but isn't clear to me if newer pandas versions were causing problems for older dask-cuda versions

Reply:
It looks like the upper bound on Pandas in RAPIDS has been around for a while. When upgrading to Pandas 1.5, the pinning was bumped to this upper bound ( https://github.com/rapidsai/integration/pull/534 )

That said, I don't why Dask-CUDA needs the Pandas pin necessarily

Maybe one explanation is RAPIDS isn't compatible with Pandas 2.0 and needs an upper bound in multiple libraries?

cc @galipremsagar (in case you have more insight 🙂)

Reply:
We have been maintaining the same upper bounds across RAPIDS to make it easier to maintain and debug any conda issues. However I don't think we have tested `dask-cuda` with pandas-2.0 yet because we are still adding `pandas-2.0` support in `cudf`, so only once we change the upper bound at `cudf` we should be able to test it in `dask-cuda`. I say we keep the pinnings consistent across RAPIDS for now and about applying the repodata hot-fix to older versions, we _might not_ need it because `cudf` bring in that tight pinning. However, if you both feel it's trivial to fix it I'm okay with it aswell.

Reply:
Thanks for the context 🙂 I'm interested in if there are specific use cases for dask-cuda that don't involve cuDF where we might be able to get away with leaving this dependency unpinned (maybe @pentschev has some thoughts?), but if those cases don't exist / we aren't testing against them agree that pinning away from pandas 2.0 is best for now.

I can open a PR applying the repodata hot-fix and we can continue discussion there - not sure if we would get some pushback since my understanding is that the repodata patches are mostly to prevent explicit failures, but I know in the past we've done stuff like this for the dask-cuda package.

Reply:
Arguably, for RAPIDS we should keep the pin that works with the rest of RAPIDS. However, this is the `conda-forge` package that gets used by the non-RAPIDS users from the community, for example, CuPy-based workflows. In here, I believe it makes sense that we follow the pins that the rest of Dask/Distributed use. Can we unpin pandas only here, while keeping the packages in `rapidsai`/`rapidsai-nightly` consistent with the remaining of RAPIDS?

Reply:
Think that would cause issues at least for this release, as my understanding is that this pinning is getting pulled from the pyproject.toml:

https://github.com/rapidsai/dask-cuda/blob/590d26ab4d61b785789ee6bacae29c37337e6703/pyproject.toml#L27

So my understanding is that even if we did pull pandas 2 with this package, it would throw an error due to the mismatch in expected/actual versions.

If we wanted to have separate pinnings for pandas between the conda-forge and RAPIDS package, I _think_ we would need the pyproject pinning to ideally be the looser of the two pinnings?

Reply:
We discussed offline, and since we don't currently test against pandas 2.0 it's probably best for now to keep the pin. If some user requests this change then we can review this decision. In any case, the plan is to have support for pandas 2.0 around August, for RAPIDS 23.08.